pkg/querier/remote_read_test.go

```diff
@@ -208,279 +325,566 @@ func TestRemoteReadHandler_Samples(t *testing.T) {
 			err = proto.Unmarshal(responseBody, &response)
 			require.NoError(t, err)
 
+			// Build expected response
+			var expectedResults []*prompb.QueryResult
+			for _, expectedResult := range queryData.expectedResults {
+				expectedResults = append(expectedResults, &prompb.QueryResult{
+					Timeseries: expectedResult.timeseries,
+				})
+			}
+
 			expected := prompb.ReadResponse{
-				Results: []*prompb.QueryResult{
-					{
-						Timeseries: queryData.expectedTimeseries,
-					},
-				},
+				Results: expectedResults,
 			}
 			require.Equal(t, expected, response)
 
-			// Ensure the time range passed down to the queryable is the expected one.
-			require.Equal(t, queryData.expectedQueriedStart, actualQueriedStart)
-			require.Equal(t, queryData.expectedQueriedEnd, actualQueriedEnd)
+			// Verify each query was called exactly once
+			for i, queryCall := range queryCalls {
+				require.Equal(t, int64(1), queryCall.Load(), "Query %d should be called exactly once", i)
+			}
 		})
 	}
 }
 
+type queryStartEnd struct {
+	start int64
+	end   int64
+}
+
+// findQueryIndexByMatchers finds the index of the query that matches the given matchers
+func findQueryIndexByMatchers(queries []*prompb.Query, matchers []*labels.Matcher) int {
+	// If there's only one query, return index 0
+	if len(queries) == 1 {
+		return 0
+	}
```

Comment:
Shouldn't we still check the matchers are what we expect them to be in this case?